docs/architecture/adr-047-extend-upgrade-plan.md

```diff
@@ -0,0 +1,206 @@
+# ADR 047: Extend Upgrade Plan
+
+## Changelog
+
+- Nov, 23, 2021: Initial Draft
+
+## Status
+
+DRAFT Not Implemented
+
+## Abstract
+
+This ADR expands the existing x/upgrade `Plan` proto message to include new fields for defining pre-run and post-run processes.
+It also defines a structure for providing downloadable artifacts involved in an upgrade.
+
+## Context
+
+The `upgrade` module in conjunction with Cosmovisor are designed to facilitate and automate a blockchain's transition from one version to another.
+
+Users submit a software upgrade governance proposal containing an upgrade `Plan`.
+The [Plan](https://github.com/cosmos/cosmos-sdk/blob/v0.44.5/proto/cosmos/upgrade/v1beta1/upgrade.proto#L12) currently contains the following fields:
+- `name`: A short string identifying the new version.
+- `height`: The chain height at which the upgrade is to be performed.
+- `info`: A string containing information about the upgrade.
+
+The `info` string can be anything.
+However, Cosmovisor will try to use the `info` field to automatically download a new version of the blockchain executable.
```

Comment:
```suggestion
The `info` string can be anything, however it is intended to specify the version of the blockchain executable to auto-download.
The name `info` is kept for backwards compatibility purposes.
```

Reply:
This `Context` section is just about how things currently are. It's not the place for the backwards compatibility note.
I'm also uncomfortable labeling the intent of the `info` field since it's currently not well defined (and unclear). Use of the field by Cosmovisor is defined, but the spec on the field in general isn't.